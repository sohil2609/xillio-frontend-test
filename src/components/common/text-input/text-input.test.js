import React from 'react';
import { shallow } from 'enzyme';
import TextInput from './text-input.component';

const setUp = (props = {}) => {
	const component = shallow(<TextInput {...props} />);
	return component;
};

describe('Text Input Component', () => {
	const mockProps = {
		value: 'xyz',
		onChange: jest.fn(),
		name: 'name',
		id: 'name',
		type: 'text'
	};
	const component = setUp(mockProps);

	it('Should test snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('Should call onChange when input is changed', () => {
		const wrapper = component.find('[data-test="textInput"]');
		wrapper.simulate('change', { target: { value: 'abc' } });
		expect(mockProps.onChange.mock.calls.length).toBe(1);
	});
});
