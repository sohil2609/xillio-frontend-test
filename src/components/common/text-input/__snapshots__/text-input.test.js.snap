// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo Item Component Should test snapshot 1`] = `ShallowWrapper {}`;

exports[`Text Input Component Should test snapshot 1`] = `ShallowWrapper {}`;
